### PR TITLE
coord: remove table granularity logging

### PR DESCRIPTION
**Pull request by mjibson**
Previously tables would check to see if they needed a manual advancing
(not triggered by `need_advance`) on each Coordinator message. While
not expensive to performance, it was confusing to know exactly what was
going on and why.

After some investigation and discussion, it seems that the original reason
for this is perhaps gone. When tables were added, there were no internal
system tables, and so something needed to advance table timestamps if
there were no table operations, which led to the logging_granularity
stuff. Now there are some system tables that are updated each second,
rendering this manual update needless (I think). Thus it seems that we
can remove the manual table advance.

If we discover some negative user experience or flakey tests then we
can add this back, but with increased understanding of why it's needed,
with improved documentation or testing to assert the behavior we need.

The comment on `logging_granularity` was for a different variable
previously removed and should never have been attached to
`logging_granularity`.

### CODE REVIEW

**Comment by mjibson** [comment]
A lotta words going around the fact that I'm not sure anyone knows what this code is needed for anymore. Let's remove it and see if anything breaks.

**Review by frankmcsherry — APPROVED** [review]
I have no idea either, so lgtm. 

I think that should there be future user stories about why this or anything like it is important we should record the information somewhere durable and searchable!

File: src/coord/src/coord.rs [hunk 1]
```diff
@@ -413,65 +409,63 @@ impl Coordinator {
         self.send_builtin_table_updates(builtin_table_updates).await;
 
         // Announce primary and foreign key relationships.
-        if self.logging_granularity.is_some() {
```
Comment (benesch) [inline src/coord/src/coord.rs:416]: Removing this `if` seems like it will likely have broken `materialized --logging-granularity=off`?

**Review by benesch — COMMENTED** [review]
No opinion on the meat of this change, but one potential problem spotted!

**Comment by benesch** [comment]
By "no opinion" I mean "seems great, but not enough context to be sure."

File: src/coord/src/coord.rs (hunk 1 above)
Comment (mjibson) [reply src/coord/src/coord.rs:416]: That option (now called `--introspection-frequency`) still worked here and disabled the logging tables, and I'm not sure what dataflow did when it received these updates? It doesn't panic for some reason I didn't trace down. Anyhow I've replaced this option with a boolean so we can disable this branch, which seems good enough for now.

**Pull request merged**